pam/gdm/protocol_test.go

```diff
@@ -0,0 +1,672 @@
+package gdm
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func reformatJSON(t *testing.T, input []byte) []byte {
+	// We can't safely compare for JSON values when generated via protobuf
+	// so we initially pass it to native implementation to make it rebuild
+	// the JSON data so that the output is more reliable.
+	// See: https://protobuf.dev/reference/go/faq/#unstable-json
+	t.Helper()
+	var js json.RawMessage
+	err := json.Unmarshal(input, &js)
+	require.NoError(t, err)
+	out, err := json.Marshal(js)
+	require.NoError(t, err)
+	return out
+}
+
+func TestGdmStructsMarshal(t *testing.T) {
+	t.Parallel()
+	tests := map[string]struct {
+		gdmData *Data
+
+		wantJSON  string
+		wantError string
```

Comment:
Can you rename this to something like `wantErrMsg`? We usually use `wantErr` as a boolean value to indicate whether or not the function should return an error and `wantErrSomething` to indicate that we want a specific error something.

Reply:
Fair, regarding to this I would have preferred to actually do `error` comparisons but, those imply defining a tons of errors or a new error kind for this purpose. I'm fine in doing this, but that's probably too much since we're not going to check for error kind at runtime, but only during tests.